swap/log.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	log "github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethersphere/swarm/network"
+)
+
+const (
+	// UndefinedAction is the default actions filter for swap logs
+	UndefinedAction string = "undefined"
+	// InitAction used when starting swap
+	InitAction string = "init"
+	// StopAction used when stopping swap
+	StopAction string = "stop"
+	// UpdateBalanceAction used when updating balances
+	UpdateBalanceAction string = "update_balance"
+	// SendChequeAction used for cheque actions
+	SendChequeAction string = "send_cheque"
+	// HandleChequeAction used for cheque actions
+	HandleChequeAction string = "handle_cheque"
+	// CashChequeAction used for cheque actions
+	CashChequeAction string = "cash_cheque"
+	// DeployChequebookAction used when deploying chequebooks
+	DeployChequebookAction string = "deploy_chequebook_contract"
+)
+
+const swapLogLevel = 3       // swapLogLevel indicates filter level of log messages
+const fileSizeLimit = 262144 // max bytes limit for splitting file in parts
+const emptyLogPath = ""      // Used when no logPath is specified for a logger
```

Comment:
Instead of defining this as a constant here, can we just do a check somewhere in a higher-level code, to verify what happens when the user does not put in a `path` for the logger. At that point, we can pass in "", if not specified by the user. From a low-level package, I would not expect that it is able to define and set the default logpath if nothing was passed in. 

I would expect the package to return an error if no `path` was passed in.

Reply:
this check is already being done in the  `setLoggerHandler` function, no? why should we move it to higher level code?

it seems to me like what you're suggesting is already the case. `logpath` is the parameter derived from user input, it's then compared to `emptyLogPath`

Reply:
The swap logger functions by doing a tee between the standard logger and the rotatingFileHandler.

So if no path is defined the default action is to continue logging on standard output instead of giving an error.

In this way, it does not affect the test packages, or the normal swarm node operation if no path is being used.